### PR TITLE
Upgrade to RN 0.54.3 and other dependencies

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Upgraded React Native to version 0.54.2 and all the javascript dependencies as well

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9892

#### Device Information
This PR was tested on: iOS 11.2 and Android 7.1.1

### CODE REVIEW

File: app/components/team_icon/team_icon.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default class TeamIcon extends React.PureComponent {
             teamIconContent = (
                 <Image
                     style={[styles.image, styleImage]}
-                    source={{uri: teamIconUrl}}
+                    source={{uri: teamIconUrl, headers: {Authorization: `Bearer ${Client4.getToken()}`}}}
```
Comment (koxen) [inline app/components/team_icon/team_icon.js]: if it's breaking why doesn't it need to be on profile pictures too

File: app/components/post_textbox/post_textbox.js [hunk 2]
```diff
@@ -170,16 +170,18 @@ export default class PostTextbox extends PureComponent {
...
-        if (contentHeight < INITIAL_HEIGHT) {
-            contentHeight = INITIAL_HEIGHT;
-        } else if (Platform.OS === 'ios') {
-            contentHeight += 5;
-        }
+        if (Platform.OS === 'android') {
+            let contentHeight = event.nativeEvent.contentSize.height;
+            if (contentHeight < INITIAL_HEIGHT) {
+                contentHeight = INITIAL_HEIGHT;
+            } else if (Platform.OS === 'ios') {
```
Comment (hmhealey) [inline app/components/post_textbox/post_textbox.js]: Why is this now all wrapped to only run on Android? If that's correct, this else if case should be removed

File: package.json [hunk 3]
```diff
@@ -16,43 +16,43 @@
     "intl": "1.2.5",
     "jail-monkey": "0.2.0",
     "mattermost-redux": "mattermost/mattermost-redux",
-    "prop-types": "15.6.0",
-    "react": "16.2.0",
+    "prop-types": "15.6.1",
+    "react": "^16.3.0-alpha.1",
```
Comment (saturninoabril) [inline package.json]: I trust that you have valid reason in stepping up early on at 16.3 even still on alpha.
I believe there's so much hype in 16.3, and I'm excited to implement some of those in the near future.

Comment (enahum) [reply package.json]: Reason being that the react native upgrade script did it automatically as is a direct dependency

**Comment by koxen** [comment]
I found a couple of regressions - most likely caused by this PR - are you able to reproduce?
(99% sure it's related to rn 0.52 upgrade)
1. the mp4 player overlay with "download" button is now incorrectly rendered; the icon is not centered
2. long post which has both opengraph preview and file attachments gets cut in the middle of file attachment; works correctly on build 91

**Comment by enahum** [comment]
Thanks @koxen I’ll take a look soon

**Comment by enahum** [comment]
This is now working fine for me

File: app/components/team_icon/team_icon.js (hunk 1 above)
Comment (hmhealey) [reply app/components/team_icon/team_icon.js]: This component is used in the share extension where the headers are not set automatically, but profile pictures are not

**Pull request merged**